tests/test_fields.py

```diff
@@ -547,6 +570,94 @@ def b(self):  # noqa: D102
         page.b
 
 
+def test_field_processors_default() -> None:
+    @attrs.define
```

Comment:
I think my comments off-GitHub might have lead you to overuse `@define`. I believe `@define` is only needed when new `__init__` parameters (page object inputs) are defined, which is not the case in most of this PR.

The thing I thought worth testing on this front is that, if you have:

```
class BasePage(ItemPage):
    class Processors:
        ...
```

You can later on use `BasePage.Processors`, i.e. `attrs` does not somehow hide the nested class. And if it *does* hide it, we may want to cover it in the docs, and I wonder if using `@define` in that case would solve the issue. But it may just work without `@define`, and I am being too paranoid about `attrs`.

Reply:
I think I may have confused myself, I remembered it as "new fields need `@attrs.define`", I've even seen that some test worked without it...

Reply:
Shouldn't attrs.define be removed?

Reply:
New dependencies need attrs.define; new fields shouldn't need it.

Reply:
I think I've kept this one in f64385f38a25fabf83c89823f020b8d35d3933a3 to show that the comment in this thread  ("The thing I thought worth testing on this front [...]") indeed works.